### PR TITLE
Enable usage of TTreeCache in Fireworks

### PR DESCRIPTION
**Pull request by osschar**
Port from 9.2.7 to 9.4.0. See #20118 for discussion.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
Please test

**Comment by Dr15Jones** [comment]
+1

File: Fireworks/Core/src/FWFileEntry.cc [hunk 1]
```diff
@@ -98,26 +107,36 @@ void FWFileEntry::openFile(bool checkVersion)
...
-   //TTreeCache::SetLearnEntries(2);
-   //m_eventTree->SetCacheSize(10*1024*1024);
-   //TTreeCache *tc = (TTreeCache*) m_file->GetCacheRead();
-   //tc->AddBranch(m_event->auxBranch_,kTRUE);
-   //tc->StartLearningPhase();
+   // Initialize caching, this helps also in the case of local file.
+   if (FWTTreeCache::IsLogging())
+     printf("FWFileEntry::openFile enabling FWTTreeCache for file class '%s'.", m_file->ClassName());
+
+   auto tc = new FWTTreeCache(m_eventTree, FWTTreeCache::GetDefaultCacheSize());
```
Comment (davidlange6) [inline Fireworks/Core/src/FWFileEntry.cc]: hi @osschar - is this pointer cleaned up somewhere?

**Comment by osschar** [comment]
Yes, ownership is transferred to TFile on the next line, 122.

**Comment by davidlange6** [comment]
+1

**Pull request merged**